src/io/flutter/view/FlutterView.java

```diff
@@ -1,671 +1,687 @@
+      toolbarGroup.add(registerAction(new TogglePlatformAction(app)));
+      toolbarGroup.add(registerAction(new PerformanceOverlayAction(app)));
+      toolbarGroup.addSeparator();
+      toolbarGroup.add(registerAction(new OpenTimelineViewAction(app)));
+      toolbarGroup.add(registerAction(new OpenObservatoryAction(app)));
+      toolbarGroup.addSeparator();
+      toolbarGroup.add(new OverflowActionsAction(this, app));
+      return toolbarGroup;
+    }
 
-  FlutterViewAction registerAction(FlutterViewAction action) {
-    getOrCreateStateForApp(action.app).flutterViewActions.add(action);
-    return action;
-  }
+    FlutterViewAction registerAction(FlutterViewAction action) {
+      getOrCreateStateForApp(action.app).flutterViewActions.add(action);
+      return action;
+    }
 
-  private PerAppState getStateForApp(FlutterApp app) {
-    return perAppViewState.get(app);
-  }
+    private PerAppState getStateForApp(FlutterApp app) {
+      return perAppViewState.get(app);
+    }
 
-  private PerAppState getOrCreateStateForApp(FlutterApp app) {
-    PerAppState state = perAppViewState.get(app);
-    if (state == null) {
-      state = new PerAppState();
-      perAppViewState.put(app, state);
+    private PerAppState getOrCreateStateForApp(FlutterApp app) {
+      PerAppState state = perAppViewState.get(app);
+      if (state == null) {
+        state = new PerAppState();
+        perAppViewState.put(app, state);
+      }
+      return state;
     }
-    return state;
-  }
 
-  private void addInspectorPanel(String displayName,
-                                 InspectorService.FlutterTreeType treeType,
-                                 FlutterApp flutterApp,
-                                 @NotNull ToolWindow toolWindow,
-                                 DefaultActionGroup toolbarGroup,
-                                 boolean selectedContent) {
-    {
+    private void addInspector(FlutterApp app, ToolWindow toolWindow) {
       final ContentManager contentManager = toolWindow.getContentManager();
       final ContentFactory contentFactory = ContentFactory.SERVICE.getInstance();
-      final Content content = contentFactory.createContent(null, displayName, false);
-      content.setCloseable(true);
-      final SimpleToolWindowPanel windowPanel = new SimpleToolWindowPanel(true, true);
-      content.setComponent(windowPanel);
-
-      final InspectorPanel inspectorPanel = new InspectorPanel(this, flutterApp, flutterApp::isSessionActive, treeType);
-      windowPanel.setContent(inspectorPanel);
-      windowPanel.setToolbar(ActionManager.getInstance().createActionToolbar("FlutterViewToolbar", toolbarGroup, true).getComponent());
+      final SimpleToolWindowPanel toolWindowPanel = new SimpleToolWindowPanel(true);
+      final JBRunnerTabs tabs = new JBRunnerTabs(myProject, ActionManager.getInstance(), null, this);
+      ArrayList<FlutterDevice> existingDevices = new ArrayList<>();
```

Comment:
I think we should prefer the interfaces for the LHS (`List`).

Reply:
done